### PR TITLE
download ticket pdf button

### PR DESCRIPTION
**Pull request by IliaStepanov**
**REWIEW https://github.com/TheDoctorZee/lowcost/pull/25 FIRST PLEASE**

### CODE REVIEW

**Comment by IliaStepanov** [comment]
[[BE] ability to download pdf](https://trello.com/c/g9BbWEtv/104-be-ability-to-download-pdf)

File: src/main/java/com/epam/lowcost/controller/TicketController.java [hunk 1]
```diff
@@ -32,18 +39,41 @@ public TicketController(TicketService ticketService, PDFService pdfService, Emai
...
+
+        headers.add("Content-Disposition", String.format("attachment; filename=\"%s\"", file.getName()));
+        headers.add("Cache-Control", "no-cache, no-store, must-revalidate");
+        headers.add("Pragma", "no-cache");
+        headers.add("Expires", "0");
+
+        ResponseEntity<Object> responseEntity = ResponseEntity.ok().headers(headers).contentLength(file.length()).contentType(
+                MediaType.parseMediaType("application/txt")).body(resource);
+
+        return responseEntity;
```
Comment (EugeneGrim) [inline src/main/java/com/epam/lowcost/controller/TicketController.java]: It's better to do inline this variable:
return ResponseEntity.ok().headers(headers).contentLength(file.length()).contentType(
                MediaType.parseMediaType("application/txt")).body(resource);

Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/TicketController.java]: done

File: src/main/java/com/epam/lowcost/services/implementations/PDFService.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 public class PDFService {
 
     public void createPDF_Ticket(Ticket ticket) throws Exception {
-        PdfWriter writer = new PdfWriter(String.format("Ticket_%d.pdf", ticket.getId()));
+        PdfWriter writer = new PdfWriter(String.format("src/main/webapp/resources/tickets_pdf/Ticket_%d.pdf", ticket.getId()));
```
[the remark this answers is not in the document]
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/services/implementations/PDFService.java]: Done.

**Pull request merged**